src/main/java/frc/robot/subsystems/drive/Drive.java

```diff
@@ -0,0 +1,313 @@
+
+import com.pathplanner.lib.auto.AutoBuilder;
+import com.pathplanner.lib.pathfinding.Pathfinding;
+import com.pathplanner.lib.util.HolonomicPathFollowerConfig;
+import com.pathplanner.lib.util.PathPlannerLogging;
+import com.pathplanner.lib.util.ReplanningConfig;
+import edu.wpi.first.math.estimator.SwerveDrivePoseEstimator;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.geometry.Twist2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import frc.robot.util.LocalADStarAK;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import org.littletonrobotics.junction.AutoLogOutput;
+import org.littletonrobotics.junction.Logger;
+
+public class Drive extends SubsystemBase
+{
+	private static final double MAX_LINEAR_SPEED = Units.feetToMeters(14.5);
+	private static final double TRACK_WIDTH_X = Units.inchesToMeters(25.0);
+	private static final double TRACK_WIDTH_Y = Units.inchesToMeters(25.0);
+	private static final double DRIVE_BASE_RADIUS = Math.hypot(TRACK_WIDTH_X / 2.0, TRACK_WIDTH_Y / 2.0);
+	private static final double MAX_ANGULAR_SPEED = MAX_LINEAR_SPEED / DRIVE_BASE_RADIUS;
+
+	static final Lock odometryLock = new ReentrantLock();
+	private final GyroIO gyroIO;
+	private final GyroIOInputsAutoLogged gyroInputs = new GyroIOInputsAutoLogged();
+	private final Module[] modules = new Module[4]; // FL, FR, BL, BR
+	private final SysIdRoutine sysId;
+
+	private SwerveDriveKinematics kinematics = new SwerveDriveKinematics(getModuleTranslations());
+	private Rotation2d rawGyroRotation = new Rotation2d();
+	private SwerveModulePosition[] lastModulePositions = // For delta tracking
+			new SwerveModulePosition[]
+			{ new SwerveModulePosition(), new SwerveModulePosition(), new SwerveModulePosition(),
+					new SwerveModulePosition() };
+	private SwerveDrivePoseEstimator poseEstimator = new SwerveDrivePoseEstimator(kinematics, rawGyroRotation,
+			lastModulePositions, new Pose2d());
+
+	public Drive(GyroIO gyroIO, ModuleIO flModuleIO, ModuleIO frModuleIO, ModuleIO blModuleIO, ModuleIO brModuleIO)
+	{
+		this.gyroIO = gyroIO;
+		modules[0] = new Module(flModuleIO, 0);
+		modules[1] = new Module(frModuleIO, 1);
+		modules[2] = new Module(blModuleIO, 2);
+		modules[3] = new Module(brModuleIO, 3);
+
+		// Start threads (no-op for each if no signals have been created)
+		PhoenixOdometryThread.getInstance().start();
```

Comment:
Can we leave odometry out of this pr? Let's get to it after getting driving + logging up and running